server/fleet/hosts.go

```diff
@@ -555,6 +555,28 @@ func (h *Host) NeedsDEPEnrollment() bool {
 		h.IsDEPAssignedToFleet()
 }
 
+// IsElegibleForWindowsMDMEnrollment returns true if the host can be enrolled
+// in Fleet's Windows MDM (if Windows MDM was enabled).
+func (h *Host) IsElegibleForWindowsMDMEnrollment(excludedTeamNames []string) bool {
+	// TODO: double-check those conditions (esp. 3rd party?)
```

Comment:
My take on "reasonable" conditions : 
* Host is Windows
* Is osquery enrolled (we do that for mac too, not sure how it could not be osquery but that makes sense to me)
* Is NOT already enrolled in a third-party MDM 
* Is NOT already enrolled in Fleet MDM (that's basically the condition that will cause the notification to stop being sent)
* Is NOT a Windows Server (though if we don't have any row in `host_mdm` yet, we don't know - is there another way to detect this, or is this sufficient?)

And also, if those conditions are met, ensure that the host's team is not excluded from Windows MDM.

Reply:
Regarding Windows Server, this is mentioned in #12260 : 

> Fleetd programmatic enrollment should not be done on Windows Servers. There should be logic that detects if running on a Windows Workstation or Windows Server, and only triggers enrollment if not running on Windows Server. This can be done by reading the registry key

So how I guess it would work is that if at first we don't know on the server-side, we'll send the notification (Needs enrollment = true), but on the device itself we'll check and see that it's a server (via the registry key) and won't enroll. At some point (correct me if I'm wrong) the server will send its osquery information and we will store the `host_mdm` data with `is_server` set to true, and from this point we would stop sending the notfication to `true`.

However, I'm not sure if we store the host MDM info when it is not enrolled, in which case we'd keep sending the notification to `true` forever, which is not great.

Reply:
cc: @marcosd4h

Reply:
I'm still getting familiarized with MacOS MDM implementation. Question: What would be the impact of having the Windows server check on the client side? In this scenario, we would mark the host as eligible for MDM enrollment, but the client will never perform the enrollment if it is a Windows server host. Would this impact a counter indicating pending mdm enrollment devices?

Reply:
I think we generally try to do any check of that nature with data that's stored in the database, for the exact reason you mention: it's also helpful to drive other logic (like the counts, but it's not unique to that)

@marcosd4h my takeaway from this is:

1. Could you double check if the conditions that Martin outlined in the first message are good enough?
2. We need to ingest and store in the db some kind of indicator that tells us if the device is a server. We fix this by doing what Sarah suggested above or by storing this attribute in a different table maybe? I think it grants a separate discussion in a follow up ticket.

Reply:
@mna regarding the points that Roberto mentioned, 
> Could you double check if the conditions that Martin outlined in the first message are good enough?

I think the conditions you defined are great and that we should go with that

> We need to ingest and store in the db some kind of indicator that tells us if the device is a server regardless of the device enrollment status.

I noticed that the registry table `InstallationType` is already being ingested and could be used to determine if host is server or workstation - see [here](https://github.com/fleetdm/fleet/blob/2c45d25ad397795ed621d1d41b7c1807360e1317/server/service/osquery_utils/queries.go#L457) and [here](https://stackoverflow.com/questions/11655364/how-do-you-find-out-the-difference-between-workstations-and-servers-on-the-netwo). We can use a separate ticket to handle this work so it does not impact this story

Reply:
@marcosd4h Thanks, I'll remove the TODO and we can go ahead with those conditions.

Regarding Windows Server, it is returned by the osquery query and ingested, but the concern is that there's a condition where if the host is unenrolled (no server URL), we delete the `host_mdm` row, so we lose the `is_server` boolean in that table: https://github.com/fleetdm/fleet/blob/2c45d25ad397795ed621d1d41b7c1807360e1317/server/datastore/mysql/hosts.go#L2753-L2772

I think for now it may be OK, in the follow-up ticket to trigger the enrollment (https://github.com/fleetdm/fleet/issues/12260) I implemented the check client-side to prevent the enrollment if it is a Windows Server, and cache that information so that we don't have to check constantly in the registry. Eventually, we should probably have that information server-side though (independently of MDM enrollment), as @roperzh mentions above. I don't think this should block this PR from merging.